cargo-paradedb/src/bench_hits.rs

```diff
@@ -0,0 +1,345 @@
+        tag
+    );
+    println!("*********************************************************************************\n");
+
+    let root_file_path = PathBuf::from_str("/tmp")?;
+    let single_file_path = root_file_path.join("hits.parquet").display().to_string();
+    let partitioned_dir_path = root_file_path.join("partitioned/").display().to_string();
+
+    if workload == "single" {
+        run_cmd!(echo "Using ClickBench's single Parquet file for benchmarking...")?;
+        download_and_verify(
+            "https://paradedb-benchmarks.s3.amazonaws.com/hits_5m_rows.parquet",
+            "5182ed3b0ad35137db38faac395d7f55",
+            &single_file_path,
+        )?;
+    } else if workload == "partitioned" {
+        run_cmd!(echo "Using ClickBench's one hundred partitioned Parquet files for benchmarking...")?;
+        run_cmd!(mkdir -p $partitioned_dir_path)?;
+        run_cmd!(
+           seq 0 99
+          | xargs "-P100" "-I{}" wget --no-verbose --no-check-certificate --directory-prefix $partitioned_dir_path
+            --continue "https://paradedb-benchmarks.s3.amazonaws.com/partitioned_5m_rows/hits_{}.parquet"
+        )?;
+    } else {
+        eprintln!("Invalid workload: {}", workload);
+        std::process::exit(1);
+    }
+
+    run_cmd!(echo "\nLoading dataset...")?;
+    env::set_var("PGPASSWORD", "postgres");
+
+    let conn_opts = &PgConnectOptions::from_str(url)?;
+    let mut conn = PgConnection::connect_with(conn_opts).await?;
+
+    let file_path = if workload == "single" {
+        single_file_path
+    } else {
+        partitioned_dir_path
+    };
+
+    println!("Creating foreign table from '{}'", file_path);
+
+    conn.execute(create_query_sql(&file_path).as_str())
+        .await
+        .unwrap();
+
+    run_cmd!(echo "\nRunning queries...")?;
+    run_queries(&mut conn).await?;
+
+    // run_cmd!(echo "\nPrinting disk usage...")?;
+    // if os.trim() == "Linux" {
+    //     run_cmd!(sudo docker exec paradedb du -bcs /bitnami/postgresql/data)?;
+    // } else {
+    //     run_cmd!(docker exec paradedb du -bcs /bitnami/postgresql/data)?;
+    // }
+
+    // run_cmd!(echo "\nPrinting results...")?;
+    // let file = File::open("log.txt")?;
+    // let reader = io::BufReader::new(file);
+    // let mut i = 0;
+
+    // for line in reader.lines() {
+    //     let line = line?;
+    //     if let Some(time_str) = line.split("Time: ").nth(1) {
+    //         if let Some(time_str) = time_str.split(" ms").next() {
+    //             let time: f64 = time_str.parse().unwrap_or(0.0);
+    //             if i % 3 == 0 {
+    //                 print!("[");
+    //             }
+    //             print!("{}", time / 1000.0);
+    //             if i % 3 != 2 {
+    //                 print!(",");
+    //             } else {
+    //                 println!("],");
+    //             }
+    //             i += 1;
+    //         }
+    //     }
+    // }
+    Ok(())
```

Comment:
Generating the results in the format:

```bash
    [0.450533,0.241519,0.241897],
    [0.41147,0.269899,0.273598],
    [0.5625,0.319037,0.317889],
    [1.90787,0.328827,0.327382],
    [2.24784,1.1383,1.1594],
    [2.84002,1.10211,1.10332],
    [0.446178,0.276948,0.278959],
    [0.397331,0.276682,0.272303],
    [2.8674,1.79979,1.77778],
    [3.37754,1.37852,1.38975],
    [2.13856,0.525317,0.531128],
    [2.28249,0.557421,0.555197],
    [2.91328,1.21163,1.21042],
    [6.443,2.51097,2.4843],
    [3.39634,1.64601,1.68288],
    [2.22905,1.27065,1.27691],
    [6.56673,2.95907,2.99409],
    [6.38378,2.75823,2.83892],
    [11.4375,5.98894,6.08143],
    [1.57497,0.31444,0.312506],
    [20.2374,1.8317,1.84673],
    [23.7624,2.07403,2.03667],
    [44.8532,4.39067,4.43058],
    [112.038,10.9028,10.9014],
    [6.542,0.637139,0.642362],
    [2.68897,0.566069,0.563855],
    [6.48862,0.709606,0.721308],
    [20.1354,2.61039,2.62396],
    [18.5493,10.5197,10.5404],
    [0.904712,0.636319,0.645807],
    [5.78476,1.33533,1.31656],
    [13.193,1.77434,1.76863],
    [14.0071,9.15143,9.19673],
    [21.1558,5.55113,5.48692],
    [22.4569,7.65486,7.69364],
    [2.34449,2.01481,2.06401],
    [1.54329,0.502326,0.501721],
    [0.712948,0.346445,0.349072],
    [1.63,0.385822,0.384043],
    [3.55532,0.943702,0.937631],
    [0.52862,0.299549,0.295918],
    [0.491267,0.291998,0.290557],
    [0.461201,0.306834,0.303907]
```

Would be very appreciated as this is what ClickBench expects

Reply:
On second thought, we can just leave the formatting to ClickBench's bash script, that's fine too. So long as the results can be written to some file from which they can be read? ClickBench script works as such:

```bash
cat log.txt | grep -oP 'Time: \d+\.\d+ ms' | sed -r -e 's/Time: ([0-9]+\.[0-9]+) ms/\1/' |
    awk '{ if (i % 3 == 0) { printf "[" }; printf $1 / 1000; if (i % 3 != 2) { printf "," } else { print "]," }; ++i; }'
```